dsl/src/main/scala/org/http4s/dsl/impl/ResponseGenerator.scala

```diff
@@ -119,6 +119,32 @@ trait WwwAuthenticateResponseGenerator[F[_]] extends Any with ResponseGenerator
   }
 }
 
+/** Helper for the generation of a [[org.http4s.Response]] which must contain
+  * an Allow header and may contain a body.
+  *
+  * A 405 status MUST contain an `Allow` header, which
+  * distinguishes this from other `ResponseGenerator`s.
+  */
+trait AllowResponseGenerator[F[_]] extends Any with ResponseGenerator {
+
+  def apply(allow: Allow, headers: Header*)(implicit F: Applicative[F]): F[Response[F]] =
```

Comment:
is an empty list/array valid here?

Reply:
What do you mean by valid? Do you mean valid with respect to the Http RFC? 

If you meant whether the method can be applied with no `Header`, except for the `Allow` one, then you can see in the changes to `ExampleService` below that this is the case.

Reply:
nvm, I misunderstood how this is supposed to work :)